crates/router/src/connector/cybersource.rs

```diff
@@ -384,7 +394,17 @@ impl
         req: &types::SetupMandateRouterData,
         _connectors: &settings::Connectors,
     ) -> CustomResult<RequestContent, errors::ConnectorError> {
-        let connector_req = cybersource::CybersourceZeroMandateRequest::try_from(req)?;
+        let minor_amount =
+            req.request
+                .minor_amount
+                .ok_or(errors::ConnectorError::MissingRequiredField {
+                    field_name: "minor_amount",
+                })?;
+        let amount = convert_amount(self.amount_converter, minor_amount, req.request.currency)?;
+        let connector_router_data = cybersource::CybersourceRouterData::from((amount, req));
+
+        let connector_req =
+            cybersource::CybersourceZeroMandateRequest::try_from(&connector_router_data)?;
```

Comment:
This change is not required because amount is always 0 for setup mandate flow. Please revert this change.